lib/src/value/color.ts

```diff
@@ -3,371 +3,1112 @@
+      return 'display-p3';
+    case 'prophoto':
+      return 'prophoto-rgb';
+  }
+  return space as KnownColorSpace;
+}
+
+// @TODO For some spaces (e.g. Lab and Oklab), ColorJS only accepts `l` and not
+// `lightness` as a channel name. Maybe a bug?
+/**
+ * Normalize discrepancies between Sass channel names and ColorJS channel ids.
+ */
+function encodeChannelForColorJs(channel: ChannelName): string {
+  if (channel === 'lightness') return 'l';
+  return channel;
+}
+
+/**
+ * Implement our own check of channel name validity for a given space, because
+ * ColorJS allows e.g. `b` for either `blue` or `blackness` or `b` channels.
+ */
+function validateChannelInSpace(
+  channel: ChannelName,
+  space: KnownColorSpace
+): void {
+  if (channel === 'alpha') return;
+  let valid = false;
+  switch (space) {
+    case 'rgb':
+    case 'srgb':
+    case 'srgb-linear':
+    case 'display-p3':
+    case 'a98-rgb':
+    case 'prophoto-rgb':
+    case 'rec2020':
+      valid = ['red', 'green', 'blue'].includes(channel);
+      break;
+    case 'hsl':
+      valid = ['hue', 'saturation', 'lightness'].includes(channel);
+      break;
+    case 'hwb':
+      valid = ['hue', 'whiteness', 'blackness'].includes(channel);
+      break;
+    case 'lab':
+    case 'oklab':
+      valid = ['lightness', 'a', 'b'].includes(channel);
+      break;
+    case 'lch':
+    case 'oklch':
+      valid = ['lightness', 'chroma', 'hue'].includes(channel);
+      break;
+    case 'xyz':
+    case 'xyz-d65':
+    case 'xyz-d50':
+      valid = ['x', 'y', 'z'].includes(channel);
+      break;
+  }
+  if (!valid) {
+    throw valueError(
+      `Unknown channel name "${channel}" for color space "${space}".`
+    );
+  }
+}
+
+/** Determine whether the given space is a polar color space. */
+function isPolarColorSpace(space: KnownColorSpace): space is PolarColorSpace {
+  switch (space) {
+    case 'hsl':
+    case 'hwb':
+    case 'lch':
+    case 'oklch':
+      return true;
+    default:
+      return false;
+  }
+}
+
+/**
+ * Normalize between ColorJS coordinates (which use `NaN`) and Sass Color
+ * coordinates (which use `null`).
+ */
+function getCoordsFromColor(
+  coords: [number, number, number],
+  isRgb = false
+): [number | null, number | null, number | null] {
+  let newCoords: [number | null, number | null, number | null] = coords;
+  if (isRgb) {
+    newCoords = (newCoords as [number, number, number]).map(coordToRgb) as [
+      number,
+      number,
+      number,
+    ];
+  }
+  return (newCoords as [number, number, number]).map(NaNtoNull) as [
+    number | null,
+    number | null,
+    number | null,
+  ];
+}
+
+function propertyIsSet(val: undefined | null | number): val is number | null {
+  return val === null || typeof val === 'number';
+}
+
+function checkChangeDeprecations(
+  options: {
+    [key in ChannelName]?: number | null;
+  },
+  channels: ChannelName[]
+) {
+  if (options.alpha === null) {
+    emitNullAlphaDeprecation();
+  }
```

Comment:
Style nit: Put this condition on one line. Same several places below.

Reply:
Addressed in https://github.com/sass/embedded-host-node/pull/259/commits/18ab10ee84ce3b3321b8a8b98f52c7fb79650653